web/pdf_page_view.js

```diff
@@ -104,42 +179,255 @@ const DEFAULT_LAYER_PROPERTIES =
 
 const LAYERS_ORDER = new Map([
   ["canvasWrapper", 0],
-  ["textLayer", 1],
-  ["annotationLayer", 2],
-  ["annotationEditorLayer", 3],
-  ["xfaLayer", 3],
+  ["detailLayer", 1],
+  ["textLayer", 2],
+  ["annotationLayer", 3],
+  ["annotationEditorLayer", 4],
+  ["xfaLayer", 4],
 ]);
 
+class PDFPageViewBase {
+  #enableHWA = false;
+
+  #loadingId = null;
+
+  #renderError = null;
+
+  #renderingState = RenderingStates.INITIAL;
+
+  #showCanvas = null;
+
+  canvas = null;
+
+  canvasWrapper = null;
+
+  /** @type {null | HTMLDivElement} */
+  div = null;
+
+  eventBus = null;
+
+  id = null;
+
+  pageColors = null;
+
+  renderingQueue = null;
+
+  renderTask = null;
+
+  resume = null;
+
+  constructor(options) {
+    this.#enableHWA =
+      #enableHWA in options ? options.#enableHWA : options.enableHWA || false;
```

Comment:
Can this be simplified e.g. as follows?
```suggestion
    this.#enableHWA = options.#enableHWA ?? (options.enableHWA || false);
```

Reply:
Unfortunately no: if `#enableHWA` is not there (in the `super()` call from the `PDFPageView` constructor), `options.#enableHWA` will throw.